sql/expression/function/json/json_search.go

```diff
@@ -0,0 +1,327 @@
+// search_str, or path arguments are NULL; no path exists within the document; or search_str is not found. An error
+// occurs if the json_doc argument is not a valid JSON document, any path argument is not a valid path expression,
+// one_or_all is not 'one' or 'all', or escape_char is not a constant expression.
+// The one_or_all argument affects the search as follows:
+//   - 'one': The search terminates after the first match and returns one path string. It is undefined which match is
+//     considered first.
+//   - 'all': The search returns all matching path strings such that no duplicate paths are included. If there are
+//     multiple strings, they are autowrapped as an array. The order of the array elements is undefined.
+//
+// Within the search_str search string argument, the % and _ characters work as for the LIKE operator: % matches any
+// number of characters (including zero characters), and _ matches exactly one character.
+//
+// To specify a literal % or _ character in the search string, precede it by the escape character. The default is \ if
+// the escape_char argument is missing or NULL. Otherwise, escape_char must be a constant that is empty or one character.
+// For more information about matching and escape character behavior, see the description of LIKE in Section 12.8.1,
+// “String Comparison Functions and Operators”: https://dev.mysql.com/doc/refman/8.0/en/string-comparison-functions.html
+// For escape character handling, a difference from the LIKE behavior is that the escape character for JSON_SEARCH()
+// must evaluate to a constant at compile time, not just at execution time. For example, if JSON_SEARCH() is used in a
+// prepared statement and the escape_char argument is supplied using a ? parameter, the parameter value might be
+// constant at execution time, but is not at compile time.
+//
+// https://dev.mysql.com/doc/refman/8.0/en/json-search-functions.html#function_json-search
+type JSONSearch struct {
+	JSON     sql.Expression
+	OneOrAll sql.Expression
+	Search   sql.Expression
+	Escape   sql.Expression
+	Path     sql.Expression
+}
+
+var errOneOrAll = fmt.Errorf("the oneOrAll argument to json_search may take these values: 'one' or 'all'")
+var errBadEscape = fmt.Errorf("incorrect arguments to ESCAPE")
+
+var _ sql.FunctionExpression = &JSONSearch{}
+
+// NewJSONSearch creates a new NewJSONSearch function.
+func NewJSONSearch(args ...sql.Expression) (sql.Expression, error) {
+	switch len(args) {
+	case 3:
+		return &JSONSearch{
+			JSON:     args[0],
+			OneOrAll: args[1],
+			Search:   args[2],
+		}, nil
+	case 4:
+		return &JSONSearch{
+			JSON:     args[0],
+			OneOrAll: args[1],
+			Search:   args[2],
+			Escape:   args[3],
+		}, nil
+	case 5:
+		return &JSONSearch{
+			JSON:     args[0],
+			OneOrAll: args[1],
+			Search:   args[2],
+			Escape:   args[3],
+			Path:     args[4],
+		}, nil
+	default:
+		return nil, sql.ErrInvalidArgumentNumber.New("JSON_SEARCH", "3 to 5", len(args))
```

Comment:
It looks like MySQL allows the `path` argument to be specified multiple times, so we should grab all params at position 5+ and make them paths that we iterate over. 

Here's an example I tried out from script_queries.go data you created:
```
select i, json_search(j, 'all', 'def') from t order by i;
+---+--------------------------------------------------+
| i | json_search(j, 'all', 'abc', NULL, "$.a", "$.b") |
+---+--------------------------------------------------+
| 0 | "$.a"                                            |
| 1 | "$.b"                                            |
| 2 | NULL                                             |
| 3 | NULL                                             |
| 4 | NULL                                             |
| 5 | NULL                                             |
+---+--------------------------------------------------+
```

Reply:
added support for multiple path arguments